framework/core/views/email/html/base.blade.php

```diff
@@ -0,0 +1,80 @@
+@inject('url', 'Flarum\Http\UrlGenerator')
+
+@php
+  $primaryColor = $settings->get('theme_primary_color');
+  $secondaryColor = $settings->get('theme_secondary_color');
+@endphp
+
+<!DOCTYPE html>
+<html lang="{{ $translator->getLocale() }}">
+<head>
+    <meta charset="utf-8">
+    <title>{{ $title ?? 'Flarum Email' }}</title>
+    <style>
+        body {
+            font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
+            font-size: 14px;
+            line-height: 1.5;
+            color: #333;
+            margin: 0;
+            padding: 20px;
+        }
+        .container {
+            max-width: 600px;
+            margin: 0 auto;
+            padding: 20px;
+            border: 1px solid #e1e1e1;
+            border-radius: 5px;
+            background-color: #fff;
+        }
+        .footer {
+            margin-top: 20px;
+            font-size: 12px;
+            color: #777;
+        }
+        .content-preview {
+            background-color: #f7f7f7; /* Light gray background */
+            padding: 15px;
+            margin: 15px 0;
+            border: 1px solid #e1e1e1;
+            border-radius: 5px;
+        }
+        .signoff {
+            margin-top: 20px;
+            font-style: italic;
+        }
+    </style>
+</head>
+<body>
+
+<div class="header">
+    <div class="Header-title">
+        <a href="{{ $url->to('forum')->base() }}" id="home-link">
+            @if ($settings->get('logo_path'))
+                <img src="{{ $url->to('forum')->base() . '/assets/' . $settings->get('logo_path') }}" alt="{{ $settings->get('forum_title') }}" class="Header-logo">
+            @else
+                {{ $settings->get('forum_title') }}
+            @endif
+        </a>
+    </div>
+    @yield('header')
+</div>
+
+<div class="content">
+    <div class="greeting">
+        <p>{!! $translator->trans('core.email.greeting', ['displayName' => $username]) !!}</p>
+    </div>
+    <div class="main-content">
+        @yield('content')
+    </div>
+    <div class="signoff">
+        <p>{!! $translator->trans('core.email.signoff', ['forumTitle' => $settings->get('forum_title')]) !!}</p>
+    </div>
+</div>
```

Comment:
the separation of the greeting from each mail (an future ones) doesn't seem right, isn't it basically enforced on every email now? what if I don't want my custom email to have that, or what if i'm going for something different?

Reply:
Extracting the greeting out, means that it's now easier to translate (just the once).

Previously, the greeting had to be implemented every time, and sometimes not consistently, this way, it is consistent.

Reply:
still, we are now enforcing it on every email without the ability to remove it. imo, if we had to choose between translating it once and allowing it to be editable/custom per email, the latter makes more sense to me.

Reply:
I'm afraid I disagree here.

Having the greeting uniform over all emails, presents a consistent approach, which users can use to help in part to identify the email as genuine, as it will always greet them in the same way and their current display name.

Having the possibility of seperating this out for each email, whilst it does allow for maximum freedom, just doesn't outweigh the consistency aspect, in my mind at least

Reply:
I can imagine quite a few cases where an extension will want a mail layout without the enforced greeting. This can be solved by adding a check along the lines of `@if(isset($greeting) && !$greeting)` to this base template.

Reply:
> I can imagine quite a few cases where an extension will want a mail layout without the enforced greeting. This can be solved by adding a check along the lines of `@if(isset($greeting) && !$greeting)` to this base template.

I was about to say the same thing. And I'd highly recommend against enforcing certain patterns, so this also includes the signoff.

Remember we are building software that is extensible, that means we are not to judge whether something is necessary. Especially with rewrites we need to be careful in allowing more extensibility and customizability instead of increasing opinionatedness.

@imorland remember that client that wanted a community where anonymity is key; it would be interesting for them to be able to disable greeting and signoff.